PyInstaller/building/api.py

```diff
@@ -171,8 +177,44 @@ def assemble(self):
                 self.code_dict[entry[0]] = self.__get_code(entry[0], entry[1])
         # sort content alphabetically to support reproducible builds
         toc.sort()
+
+        # Remove leading parts of paths in code objects
+        self.code_dict = {
+            key: self._strip_paths_in_code(code)
+            for key, code in self.code_dict.items()
+        }
+
         pyz = ZlibArchiveWriter(self.name, toc, code_dict=self.code_dict, cipher=self.cipher)
 
+    def _strip_paths_in_code(self, co):
+        original_filename = os.path.normpath(co.co_filename)
+        for f in self.replace_paths:
+            if original_filename.startswith(f):
+                new_filename = original_filename[len(f):]
+                break
+
+        else:
+            return co
+
+        consts = list(co.co_consts)
+        for i in range(len(consts)):
```

Comment:
We could use `enumerate` here, as we only support 2.7 only. We could even use list comprehensions here.